bioimageio/core/_prediction_pipeline.py

```diff
@@ -40,6 +43,52 @@
 )
 
 
+@dataclass(frozen=True)
+class PredictionTensorMetadata:
+    input: Tensor
+    prediction: Tensor
+    input_blocks: List[Block]
+    prediction_blocks: List[Block]
+
+    def __post_init__(self):
+        assert len(self.input_blocks) == len(self.prediction_blocks)
+        assert is_perfect_square(self.total_blocks)
+
+    @property
+    def total_blocks(self):
+        return len(self.input_blocks)
+
+
+@dataclass(frozen=True)
+class PredictionSampleMetadata:
```

Comment:
not entirely sure about the purpose of this class.. I'll leave a series of comments and maybe then we can have a chat about it later.

1. I see how you want a structure to capture the relation of an input to an output. However,
    -  having both `input: Sample` and `input_blocks: List[SampleBlock]` seems redundant. (fine for debugging though)
    - how does `SampleBlockMeta` fit into this?
2. `Meta` in the name suggests to me that there is no actual data (but there is, this object has `input: Sample`)

Reply:
Yep, actually for the classes, I am not sure if they should be there. I just used them for  debugging, but maybe they are not a fit for the package. I just wanted a placeholder to check input, outputs, and their respective blocks. They could be easily deleted and just keep the bug fix.

The Metadata I think of it as a way of going beyond just the prediction output, and check the whole process, of what was the input, and the blocks. Maybe not the prefect name. Could be `PredictionSampleContainer`?